src/components/Choropleth.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+    <div v-if="initialising">loading..</div>
+    <div v-else>
+        <LMap ref="map" style="height: 100vh; width: 100%" @ready="updateBounds">
+            <LTileLayer
+                :url="backgroundLayer.url"
+                :attribution="backgroundLayer.attribution"
+                :max-zoom="backgroundLayer.maxZoom"
+                :min-zoom="backgroundLayer.minZoom"
+            ></LTileLayer>
+            <LGeoJson
+                v-for="f in featuresWithColours"
+                ref="featureRefs"
+                :key="featureId(f.feature)"
+                :geojson="f.feature"
+                :options="createTooltips"
+                :options-style="
+                    () => {
+                        return { ...style, fillColor: f.colour };
+                    }
+                "
+            >
+            </LGeoJson>
+        </LMap>
+    </div>
+</template>
+<script setup lang="ts">
+import { ref, computed, watch } from "vue";
+import { storeToRefs } from "pinia";
+import { GeoJSON, Layer } from "leaflet";
+import { LGeoJson, LMap, LTileLayer } from "@vue-leaflet/vue-leaflet";
+import { Feature } from "geojson";
+import { useAppStore } from "../stores/appStore";
+import { useColourScale } from "../composables/useColourScale";
+import "leaflet/dist/leaflet.css";
+
+interface FeatureWithColour {
+    feature: Feature;
+    colour: string;
+}
+
+const FEATURE_ID_PROP = "shapeID";
+const FEATURE_NAME_PROP = "shapeName";
+const FEATURE_COUNTRY_PROP = "shapeGroup";
+
+const style = {
+    className: "geojson"
+};
+
+const backgroundLayer = {
+    url: "https://server.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer/tile/{z}/{y}/{x}",
+    attribution: "Tiles &copy; Esri &mdash; Esri, DeLorme, NAVTEQ",
+    maxZoom: 10,
+    minZoom: 3
+};
+
+const map = ref<typeof LMap | null>(null);
+const featureRefs = ref<(typeof LGeoJson)[]>([]);
+
+const { selectedFeatures, selectedIndicators, loading, selectedIndicator } = storeToRefs(useAppStore());
+const { selectCountry } = useAppStore();
+
+const { colourScales, getColour } = useColourScale(selectedIndicators);
+
+const featureId = (feature: Feature) => feature.properties![FEATURE_ID_PROP];
```

Comment:
funtions should probably be action words.. eg. `getFeatureId`

Reply:
Agreed